packages/neuron-ui/src/components/DepositDialog/hooks.ts

```diff
@@ -0,0 +1,270 @@
+  isBalanceReserved,
+  walletID,
+}: {
+  suggestFeeRate: number
+  isBalanceReserved: boolean
+  walletID: string
+}): Promise<State.GeneratedTx | null> {
+  return generateDaoDepositAllTxAPI({
+    walletID,
+    feeRate: `${suggestFeeRate}`,
+    isBalanceReserved,
+  }).then(res => {
+    if (isSuccessResponse(res)) {
+      return res.result
+    }
+    throw new Error(`${typeof res.message === 'string' ? res.message : res.message.content}`)
+  })
+}
+
+export const useGenerateDaoDepositTx = ({
+  walletID,
+  isBalanceReserved,
+  depositValue,
+  suggestFeeRate,
+  showDepositDialog,
+  slidePercent,
+}: {
+  walletID: string
+  isBalanceReserved: boolean
+  depositValue: string
+  suggestFeeRate: number
+  showDepositDialog: boolean
+  slidePercent: number
+}) => {
+  const timer = useRef<ReturnType<typeof setTimeout>>()
+  const [errorMessage, setErrorMessage] = useState('')
+  const [maxDepositValue, setMaxDepositValue] = useState<string | undefined>()
+  const [t] = useTranslation()
+  const dispatch = useDispatch()
+  const clearGeneratedTx = useClearGeneratedTx()
+  const isDepositAll = useMemo(() => slidePercent === PERCENT_100, [slidePercent])
+  useEffect(() => {
+    clearTimeout(timer.current)
+    if (!showDepositDialog) {
+      return
+    }
+    timer.current = setTimeout(() => {
+      setErrorMessage('')
+      const errorDepositValue = checkDepositValue(depositValue, t)
+      if (errorDepositValue) {
+        clearGeneratedTx()
+        setErrorMessage(errorDepositValue)
+        return
+      }
+
+      const generateDaoDepositResult: Promise<State.GeneratedTx | null> = isDepositAll
+        ? generateDaoDepositAllTx({ walletID, isBalanceReserved, suggestFeeRate })
+        : generateDaoDepositTx({ walletID, capacity: CKBToShannonFormatter(depositValue), suggestFeeRate, t })
+      generateDaoDepositResult
+        .then(res => {
+          dispatch({
+            type: AppActions.UpdateGeneratedTx,
+            payload: res,
+          })
+          if (isDepositAll) {
+            setMaxDepositValue(shannonToCKBFormatter(res?.outputs[0]?.capacity ?? '0', false, ''))
+            if (!isBalanceReserved) {
+              setErrorMessage(t('messages.remain-ckb-for-withdraw'))
+            }
+          }
+        })
+        .catch((err: unknown) => {
+          clearGeneratedTx()
+          setErrorMessage(err instanceof Error ? err.message : '')
+        })
+    })
+  }, [
+    clearGeneratedTx,
+    dispatch,
+    walletID,
+    t,
+    setErrorMessage,
+    isBalanceReserved,
+    depositValue,
+    suggestFeeRate,
+    showDepositDialog,
+    isDepositAll,
+  ])
+  return {
+    errorMessage,
+    maxDepositValue: isDepositAll ? maxDepositValue ?? depositValue : null,
+  }
+}
+
+function calculatePercent(amount: string, total: string) {
+  if (!total || total === '0') return 0
+  return +((BigInt(PERCENT_100) * BigInt(amount)) / BigInt(total)).toString()
+}
+
+export const useDepositValue = (balance: string) => {
```

Comment:
Maybe it could be `return ((BigInt(PERCENT_100) * BigInt(amount)) / BigInt(total)).toNumber()`

Reply:
`BigInt` does not have `toNumber` function.